torch/_guards.py

```diff
@@ -394,6 +394,19 @@ class TracingContext:
     will return None.
     """
 
+    class TrainStepContext:
```

Comment:
Any context added needs to respect copy/restore.

Reply:
could you help me write a comment that i could add to the TracingContext code that makes it more apparent what that means and how to do it?  Also maybe a test that confirms that TracingContexts are restorable?